File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::TNGDI8;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your TNG DI8

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $di8 = Tinkerforge::TNGDI8->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current values
my $values = $di8->get_values();

print "Channel 0: " . @{$values}[0] . "\n";
print "Channel 1: " . @{$values}[1] . "\n";
print "Channel 2: " . @{$values}[2] . "\n";
print "Channel 3: " . @{$values}[3] . "\n";
print "Channel 4: " . @{$values}[4] . "\n";
print "Channel 5: " . @{$values}[5] . "\n";
print "Channel 6: " . @{$values}[6] . "\n";
print "Channel 7: " . @{$values}[7] . "\n";

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/tng_di8'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your TNG DI8

ipcon = IPConnection.new # Create IP connection
di8 = TNGDI8.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current values
values = di8.get_values

puts "Channel 0: #{values[0]}"
puts "Channel 1: #{values[1]}"
puts "Channel 2: #{values[2]}"
puts "Channel 3: #{values[3]}"
puts "Channel 4: #{values[4]}"
puts "Channel 5: #{values[5]}"
puts "Channel 6: #{values[6]}"
puts "Channel 7: #{values[7]}"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
